### ISSUE TITLE
[Feature] Running single test

### ISSUE THREAD

**Issue by Venryx** [psiho/jsbench-me#22]
It would be nice if there were a way to only rerun a single test.

Usually when I make changes, only one of the tests is changed (to fix a typo or what-not), which means it's a waste of time to have all the other tests run as well.

**Comment by psiho** [comment]
Benchmark.js library really run tests and believe it or not I'm not too good at understanding what happens under the hood. However, as I see it, and I might be wrong, individual test might be not reliable. If your PC starts some other process in the background after the first run, then your test run during that process might appear slower than it would have been if run at the same time as the first one. So I personally wouldn't trust individual test runs for comparison.

However, from usability perspective, if you just want a quick check if your code is ok and/or to check for some significant differences of speed, then this feature would help. But still, I'd like to stop people misusing it somehow and getting bad comparison results.

I'm considering this for v1. Any comments suggestions are welcome.

**Comment by Venryx** [comment]
> However, from usability perspective, if you just want a quick check if your code is ok and/or to check for some significant differences of speed, then this feature would help. But still, I'd like to stop people misusing it somehow and getting bad comparison results.

I think as long as you include some sort of warning message (after pressing the "rerun just this test" button), it should be fine.

Perhaps something like: "For best comparison accuracy, rerun the tests at the same time"

**Comment by psiho** [comment]
@Venryx ... a little help maybe with your usage scenario? I must say I don't use single runs in my workflow so I'd like to know more what you and others prefer. Bottom line, comparison id not reliable unles tests run all at the same time, but running a single one can be useful. Now, Q how to make it more useful.

My current (working) scenario is that once you run single test, all other test go to "pending..." stage. This also deletes all prior test results and comparisons though. Now, it seems ok for me, as I can only imagine single test runs would be to test if code is functioning ok, but not really to benchmark it and compare it. But I'd like to know more about your usage scenario.

Maybe you prefer to put other test into "separate run" status, keeping old results and running new comparison on all tests (%slower, fastest, slowest)? This is doable, but looks too good and so easily mistaken for valid suite test, and only complete suite test is really valid. Then some message should be added, but it might nag people using it to test code and understanding the issue. 

So, some open questions. Again, your (and others) usage scenarios would really help to make a decission.

**Comment by Venryx** [comment]
The reason I would like to be able to rerun a single test, even though it's not "valid" as a reliable comparison with the other run-at-the-same-time tests, is pretty general: it's for quick iteration.

If you have 5 approaches to solve a problem, and running each one takes 10 seconds, that means you have to wait 50 seconds each time you make a tiny change in one of the tests (well, if it's at the end), to try to make it faster, fix a bug, etc. This is _terrible_ productivity-wise. So, while rapidly making changes to the tests, I would like a way to rerun just that single test I just modified.

Yes, the results are not 100% valid/comparable with the other tests. But nonetheless it provides a _ballpark_ estimate of its relative runtime, which is often all you need while making rapid changes (ie. if something suddenly takes 3 times as long to run, it's actually because it's slower, not just differences between runs). Also, even apart from timing results, this lets you quickly detect errors in modified individual tests.

So basically, I get what you're saying about the results not being 100% valid/comparable when you rerun just a single one, but the productivity benefits while rapidly iterating are huge, and warrant adding it as an option. (and, a message can just be added to make clear that, for final comparisons, you should rerun them all at once)

**Comment by psiho** [comment]
Great. I get it. I'll sleep on it, to find a way not to nag users with messages but also to make it possible (and clear what it means).

In the meantime, I decided to post test location. It can be helpful for last minute feedback like this.

https://prerelease--jsbenchme-frontend.netlify.app/

... As said, I don't have really good public test workflow so this URL is unreliable. I'm working daily on frontend/backend which might break it. Test location is separate from live data (both users and test) although user registration/login should start working as soon as my AWS SES goes out of sanbox, which I requested yeasterday.
